be/src/exprs/agg/window.h

```diff
@@ -214,6 +214,70 @@ class DenseRankWindowFunction final : public WindowFunction<DenseRankState> {
     std::string get_name() const override { return "dense_rank"; }
 };
 
+// The NTILE window function divides ordered rows in the partition into `num_buckets` ranked groups
+// of as equal size as possible and returns the group id of each row starting from 1.
+//
+// It can not been used with the windowing clause.
+// And for the implementation, it uses `ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW`.
+//
+// The size of buckets could be `num_partition_rows/num_buckets` (small bucket)
+// or `num_partition_rows/num_buckets+1` (large bucket).
+// The top `num_partition_rows%num_buckets` buckets are the large buckets.
+struct NtileState {
+    int64_t num_buckets = 0;
+
+    int64_t large_bucket_size = -1;
+    int64_t small_bucket_size = -1;
+
+    int64_t num_large_buckets = -1;
+    int64_t num_large_bucket_rows = -1;
+
+    // Start from 0.
+    int64_t cur_position = -1;
+};
+
+class NtileWindowFunction final : public WindowFunction<NtileState> {
+    void reset(FunctionContext* ctx, const Columns& args, AggDataPtr __restrict state) const override {
+        this->data(state).num_buckets = args[0].get()->get(0).get_int64();
```

Comment:
Why not just `args[0]->get(0).get_int64()` ?

Reply:
Fixed.